services/rclone-http/default.nix

```diff
@@ -0,0 +1,75 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.services.rclone-http;
+
+  addr =
+    if cfg.withCaddy
```

Comment:
This introduces a slightly odd dependency / knowledge of another service that might be running too.

We can probably drop the boolean for now, in favor of a "listenAddress" module option.
On the caddy site, we can define a very precise `systemd.services.rclone-http.serviceConfig.ExecStartPost` where we allow the caddy user to access this.

And once we make use this much more frequently, we can come up with a more generic module option here (like "users allowed to access this socket" or something)